doc/std/D2175.md

```diff
@@ -0,0 +1,3182 @@
+is requested so that we can actively call `CancelIoEx()`.
+
+Note that the OS may or may not be able to actually cancel the operation depending
+on how far along the I/O operation has progressed (there may already be a completion
+event sitting the I/O completion port's queue) and depending on the capabilities
+of the I/O device.
+
+## Structured Concurrency
+
+One of the design principles that has been guiding the design of sender/receiver,
+coroutines and of sender-based concurrency algorithms is that of "structured concurrency".
+
+The introduction of structured control-flow, which provided composable
+control-flow structures such as `if/else`, `do/while` and `switch` blocks, made
+it easier to reason about control-flow compared to `goto`. These control-flow
+constructs can be composed arbitrarily to create complex logic and algorithms.
+Structured control-flow also tied control-flow to program scopes which made
+it easier to visually see the structure in the control-flow of your program.
+
+Similarly, the introduction of structured-lifetime in C++, which tied lifetime
+of objects to program scopes through use of destructors, ensured that
+resources were cleaned up at the end of scopes. This made it much easier to
+reason about the lifetime of objects and write types and programs that 
+can enlist the help of the compiler to ensure that resources are cleaned up,
+even in the presence of complicated control flow - like exceptions.
+
+The combination of structured lifetime and structured control-flow and tying them
+both to the same program scopes makes it much easier to reason about the code.
+
+> Structured concurrency revolves around the idea that (potentially) concurrent
+> execution can be treated as a resource with a lifetime, much like how objects
+> can have lifetimes.
+
+A potentially concurrent operation can be actively using resources it is given a
+reference to, and so the basic principle is that we need to make sure that the
+end of the lifetime of the concurrent operation "happens before" the destruction
+of any resources it might be accessing.
+
+The way do this is by ensuring that all potentially concurrent operations that
+might be accessing a given resource are "joined" before that resource is destroyed
+(i.e. when it goes out of scope).
+
+If we fail to join a potentially concurrent operation before resources it's using
+goes out of scope then the program has undefined behaviour. Further, if we create
+a detached operation (work that can _never_ be joined) then this makes it very
+difficult, and in some cases impossible, to know when it will be safe to destroy
+resources passed by-reference to that operation.
+
+Fire and forget interfaces, such as the proposed `std::execution::execute()`
+from P0443R14, often require the programmer to manually ensure that they do
+some extra work at the end of the operation to signal completion so that the
+work can be joined.
+
```

Comment:
I was looking for something stronger to draw attention to the ad-hoc and non-uniform mechanisms that are built to signal completion of work when using fire-and-forget interfaces like execute